include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h

```diff
@@ -3532,13 +3533,21 @@ namespace internal
       Assert((dynamic_cast<const FE_DGQ<dim, spacedim> *>(
                 &dof_handler.get_fe()) != nullptr) ||
                (dynamic_cast<const FE_Q<dim, spacedim> *>(
+                  &dof_handler.get_fe()) != nullptr) ||
+               (dynamic_cast<const FE_SimplexP<dim, spacedim> *>(
+                  &dof_handler.get_fe()) != nullptr) ||
+               (dynamic_cast<const FE_SimplexDGP<dim, spacedim> *>(
                   &dof_handler.get_fe()) != nullptr),
-             ExcMessage("Function expects FE_DGQ of FE_Q in dof_handler."));
+             ExcMessage("Function expects FE_DGQ or FE_Q in dof_handler."));
```

Comment:
Message is out-dated. This reminds me that we really should implement a proper detection mechanism at some point via #9544, rather than accumulating yet another workaround.